README.md

```diff
@@ -1,19 +1,160 @@
+continuously under conditions where there is no bound on message-delivery delay
+and under the assumption that there is a significant probability of malicious
+behavior, making it an excellent fit for blockchain-related applications.
+For more information, check [the paper][paper-link]
+
+This repository contains a Rust implementation of Aleph that offers a convenient
+API enabling seamless application to various problems. The prime application of
+the repository is the consensus engine (sometimes called the "finality gadget")
+of the [Aleph Zero blockchain][aleph-node-link].
+
+### Detailed documentation
+
+If the crate's [documentation][docs-link] seems to be not comprehensive enough,
+please refer to the [detailed version][reference-link].
+
+### Implementation status
+
+- current version is asynchronous, so it's driven by consensus events as opposed
+  to some clock ticks
+- while being asynchronous, the performance is still optimal in partially
+  synchronous environment
+- guaranteed safety even in asynchronous environment
+- BFT - secure if less than one third of the committee is malicious
+- secure against fork bombs, for details see [the paper][paper-link]
+- network overhead optimized to not send all parents hashes but a bitmap and a control hash
+- thorough testing, including malicious scenarios, and high code coverage
+
+### Future work
+
+- Asynchronous liveness is an important theoretical property and there is a lot of technical
+  sophistication that comes in the design of Aleph in order to achieve it, however on the practical
+  side there is still little evidence that performing such attacks against liveness in real-world
+  scenarios is possible. Still, no matter how unlikely such attacks might be, we take them very
+  seriously and plan to add randomness to AlephBFT in one of the future releases. We decided to go
+  for a version without randomness first, as it gives an incredibly simple and at the same time
+  secure and robust BFT consensus protocol. Adding randomness introduces some complexity into the
+  protocol, so it makes sense to add it on top of a well-tested, working product. The API of the
+  protocol will not change and we will make the use of randomness configurable.
+- We see a big value in keeping a critical piece of code such as a consensus protocol as
+  self-contained as possible, so we would like to get rid of the only major dependency -
+  `parity-scale-codec`
+
+### Using the crate
+
+- Import AlephBFT in your crate
+  ```
+  [dependencies]:
+  AlephBFT = "1"
+  ```
+- AlephBFT requires user to provide it with an implementation of the following traits:
+  - The [DataIO][dataio-link] trait is an abstraction for a component that provides data items,
+    checks availability of data items and allows to input ordered data items. `DataIO` is
+    parametrized with a `Data` generic type representing the type of items we would like to order.
+    ```
```

Comment:
Adding `rust` here may colour the syntax of examples

Reply:
That's neat! done.